python/examples/15-chit-chat-dialogue/dialogues/simple_chitchat.py

```diff
@@ -0,0 +1,101 @@
+"""Specific dialogue class for the chit-chat dialogue."""
+from uagents.experimental.dialogues import Dialogue, Node, Edge
+
+# Node definition for the dialogue states
+waiting_state = Node(
+    name="Waiting",
+    description=(
+        "This is the default state of the dialogue. Every session starts in "
+        "this state and is automatically updated once a transition to start or "
+        "continue a dialogue is triggered."
+    ),
+    starter=True,
+)  # currently not used as states are measured by the edges
```

Comment:
Would it make any sense to define nodes and edges as types that inherit from `Node` and `Edge`?
```python
class WaitingState(Node):
    name="Waiting"
    description=(
        "This is the default state of the dialogue. Every session starts in "
        "this state and is automatically updated once a transition to start or "
        "continue a dialogue is triggered."
    )
    starter=True
)  # currently not used as states are measured by the edges

class InitSession(Edge):
     name="Initiate Session"
     description="This is the transition to start the dialogue"
     parent: None
     child: InitState
```
This may not be standard practice, because I'm not sure whether there would be a use for instantiating one of these, but architecturally it seems like it could be sensible. This way you could write something like:
```python
@dialogue.on_message(model=InitiateChitChatDialogue, state_transition=InitSession)
```
Just an idea though, I haven't thought this all the way through.

Reply:
To keep things as familiar as possible we absolutely agree with this approach. Will be changed :)